### PR TITLE
#200 Npm updates

### PR DESCRIPTION
**Pull request by hunterwilliams**
Works on #200 

Just starting and detailing process here.

Added binary corruption fix because I want to see other possible breakages

9885521 - updates npm dependencies for slush. Works fine and builds a working app.
ef9fa79 - updates template npm.  gulp serve-local does **not** work.  Stops after wiredep.

### CODE REVIEW

**Comment by hunterwilliams** [comment]
current issue is any task seems to run the first subtask and then die. looking into it...

**Comment by hunterwilliams** [comment]
``gulp serve-local`` now runs. Errors on angular page though. 

``` javascript
angular.js:4411 Uncaught Error: [$injector:modulerr] Failed to instantiate module app due to:
Error: [$injector:nomod] Module 'app' is not available! You either misspelled the module name or forgot to load it. If registering a module ensure that you specify the dependencies as the second argument.
```

``gulp build`` has error about filters

**Comment by hunterwilliams** [comment]
So the new gulp-inject currently has issues when working with files with multiple inject sections just like the app's index.html. So rolling back to 1.5.0. The bug was reported 22 hours ago as of this post -> https://github.com/klei/gulp-inject/issues/135

File: app/templates/gulpfile.js [hunk 1]
```diff
@@ -254,9 +255,9 @@ gulp.task('optimize', ['inject', 'test'], function() {
 
   var assets = $.useref.assets({searchPath: './'});
```
Comment (hunterwilliams) [inline app/templates/gulpfile.js]: not sure if this line should be similarly changed as the filters that follow as restore() is called later on this

**Comment by hunterwilliams** [comment]
.... gulp still hangs during build on inject.

``gulp serve-local`` seems to serve a functioning app however

**Comment by hunterwilliams** [comment]
culprit line 283
``` javascript
282     .pipe(jslibFilter)
283     .pipe($.uglify()) // another option is to override wiredep to use min files
284     .pipe(jslibFilter.restore)
```

fix or reasoning. no idea yet..

File: app/templates/gulpfile.js [hunk 2]
```diff
@@ -269,18 +270,18 @@ gulp.task('optimize', ['inject', 'test'], function() {
...
     .pipe($.ngAnnotate({add: true}))
     .pipe($.uglify())
-    .pipe(jsAppFilter.restore())
+    .pipe(jsAppFilter.restore)
 
     // Get the vendor javascript
     .pipe(jslibFilter)
     .pipe($.uglify()) // another option is to override wiredep to use min files
-    .pipe(jslibFilter.restore())
+    .pipe(jslibFilter.restore)
```
Comment (hunterwilliams) [inline app/templates/gulpfile.js]: currently breaks it for some reason because it follows the above formats...

**Comment by grtjn** [comment]
A diff on the generated HTML/CSS could also be a good test. Take current latest, create project, run building of dist, take this branch, do same, compare dists.

I think this needs longer testing to be sure there are no side-effect further down the road. I'd like to push this into 0.2.1, even though it is good work, and looks very promissing..

**Comment by grtjn** [comment]
(open for debate..)

**Comment by hunterwilliams** [comment]
I think it should wait till 0.2.1. I just went ahead and started on it so you could toss in master after the 0.2.0 release that way people hammer on it sooner.

**Comment by grtjn** [comment]
Ok good, it is now under milestone 0.2.1..

**Comment by hunterwilliams** [comment]
so supposedly this should help find the error: 

```     .pipe($.uglify().on('error', $.util.log))```
but haven't had any success yet. Will work on later this weekend/next week if anyone doesn't give any guesses as to the problem

**Comment by grtjn** [comment]
Sounds like something to apply everywhere. Can't we catch that somewhere higher? Like on one place to catch all errors?

File: app/templates/gulpfile.js (hunk 2 above)
Comment (hunterwilliams) [inline app/templates/gulpfile.js]: actually doesn't... just takes like 3 mins to build

**Comment by hunterwilliams** [comment]
Added error printing to optimize step. Wasn't an error just takes a while ~ 3-6 mins to complete. Didn't realize it took that long in the working branch.

Going to test the app out some more but I think it's fine. Just should wait until 0.2.0 drops

**Comment by hunterwilliams** [comment]
This is now ready to merge. Nothing is listed as outdated as of the time of this post.

**Comment by grtjn** [comment]
Haven't tested yet, but do you know if this also solves these deprecation messages?
```
npm WARN deprecated gulp-karma@0.0.4: Please use Karma directly: https://github.com/karma-runner/gulp-karma
npm WARN deprecated mocha-clean@0.4.0: Mocha 2.2.5 already shortens stack traces by default.
```

**Comment by hunterwilliams** [comment]
Believe so. I know I got rid of the second deprecation message for sure.

**Comment by grtjn** [comment]
Yeah, the top one looks new, hadn't noticed it before..

**Comment by joemfb** [comment]
To resolve that warning, remove gulp-karma from the build, and just require and use karma directly in gulpfile.js. The entire package is deprecated ...

**Comment by grtjn** [comment]
Could anyone run a test with this? I think it is time to move forward with this. There is probably yet more to improve, but we could do that in a separate PR..

**Comment by grtjn** [comment]
@hunterwilliams Still around, and willing to touch up this PR to get it in a merge-able state?

**Comment by hunterwilliams** [comment]
I can give it a go sometime later this week. Waited long enough to attempt to merge it :P

**Comment by grtjn** [comment]
Still up for rebasing this? Or prefer me to resolve conflicts?

**Comment by hunterwilliams** [comment]
Sorry if you can go ahead and do it. I actually have been swamped lately

**Comment by grtjn** [comment]
I'll pull your changes, and see if I can rebase them, and open as a second PR..

**Comment by grtjn** [comment]
Created new PR #305 for this..

**Pull request closed without merging**